pkg/components/internal/testutil/common.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+
+	"github.com/kinvolk/lokomotive/pkg/components"
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+)
+
+// ConfigFromMap function takes a map and a key. If the key does not exists in that map it fails.
+func ConfigFromMap(t *testing.T, m map[string]string, k string) string {
+	ret, ok := m[k]
+	if !ok {
+		t.Fatalf("Config not found with filename: %q", k)
```

Comment:
>The caller has to look at the implementation to figure out we're doing this, and the result could be that the helper function decides to terminate all the unit tests (which is what Fatalf does).

`Fatalf` only terminates single unit test. See https://golang.org/pkg/testing/#T.FailNow for more details.

`Errorf` can be used if you have data ready and you perform N checks on it, then each check may use `Errorf`. And we use `Fatal` here to break the execution flow, as otherwise we would test on bad data. IMO `Fatalf` use here is OK.

Reply:
To stop the sub-test from proceeding I have used `Fatalf`. About returning things around in a test code makes no sense because we are already passing the test object (just like context) which can control the termination of the test.